dataland-internal-storage/src/main/kotlin/db/migration/V22__IntroduceNewFieldsInSfdr.kt

```diff
@@ -0,0 +1,42 @@
+package db.migration
+
+import db.migration.utils.DataTableEntity
+import db.migration.utils.migrateCompanyAssociatedDataOfDatatype
+import org.flywaydb.core.api.migration.BaseJavaMigration
+import org.flywaydb.core.api.migration.Context
+
+/**
+ * This migration script updates all sfdr datasets to match the new sfdr data model.
+ */
+class V22__IntroduceNewFieldsInSfdr : BaseJavaMigration() {
+
+    override fun migrate(context: Context?) {
+        migrateCompanyAssociatedDataOfDatatype(
+            context,
+            "sfdr",
+            this::migrateBoardFields,
+        )
+    }
+
+    /**
+     * Migrates sfdr data to the new sfdr data model.
+     */
+    fun migrateBoardFields(dataTableEntity: DataTableEntity) {
+        val dataset = dataTableEntity.dataJsonObject
+
+        val socialObject = dataset.getJSONObject("social") ?: return
+
+        val socialAndEmployeeMattersObject = socialObject.getJSONObject("socialAndEmployeeMatters") ?: return
+
+        val femaleBoardMembers = socialAndEmployeeMattersObject.remove("femaleBoardMembers")
+        socialAndEmployeeMattersObject.put("femaleBoardMembersSupervisoryBoard", femaleBoardMembers)
+
+        val maleBoardMembers = socialAndEmployeeMattersObject.remove("maleBoardMembers")
```

Comment:
Man braucht auf jeden Fall noch einen check ob der Key den man ersetzen möchte in dem json tatsächlich vorkommt. In den Daten auf Produktion kann es sein das Felder die erst später dazugekommen sind nicht in den DB strings enthalten sind.

Reply:
Im Prinzip sollte das ok sein. Remove gibt null zurück, wenn der key nicht exisitiert. Ich habe im neusten commit die Logik leicht verändert, dass in diesem Fall der key nicht weiter beachtet wird.